apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/NamespaceService.java

```diff
@@ -165,20 +173,29 @@ public List<NamespaceBO> findNamespaceBOs(String appId, Env env, String clusterN
       throw new BadRequestException("namespaces not exist");
     }
 
-    List<NamespaceBO> namespaceBOs = new LinkedList<>();
+    List<NamespaceBO> namespaceBOs = new CopyOnWriteArrayList<>();
+    CountDownLatch latch = new CountDownLatch(namespaces.size());
     for (NamespaceDTO namespace : namespaces) {
+      executorService.submit(() -> {
+        NamespaceBO namespaceBO;
+        try {
+          namespaceBO = transformNamespace2BO(env, namespace);
+          namespaceBOs.add(namespaceBO);
+        } catch (Exception e) {
+          LOGGER.error("parse namespace error. app id:{}, env:{}, clusterName:{}, namespace:{}",
+              appId, env, clusterName, namespace.getNamespaceName(), e);
+          throw e;
```

Comment:
I believe with the current change, the exception thrown here won't be aware in the calling thread.
We may use future.get to wait and retrieve the exception?

Reply:
Indeed, I have modified and added test cases